Added Overconfident NN regression notebook

haikookhandor opened the pull request:
#32 

- [x] Prediction on training data and future data.
- [x] MC Dropout standard deviation.
- [x] Different predictions also shown in the same plot.

haikookhandor committed:
Added Overconfident NN regression notebook

nipunbatra commented:
1. For all figures use the year as the X-axis and the CO2 concentration in pp .. as the label.
2. Remove any unneeded legends (esp. when just plotting the data)
3. See: https://www.cs.ox.ac.uk/people/yarin.gal/website/PDFs/2017_OReilly_talk.pdf where the prediction with MC Dropout looks more reasonable. In fact, we may not want to use a heteroskedastic likelihood, but just MC Dropout to capture uncertainty may be enough.
4. See: https://ml-course.github.io/master/labs/Lab%205%20-%20Bayesian%20learning for same problem with GP

haikookhandor committed:
Incorporated comments

haikookhandor commented:
1. Updated.
2. Updated.
3. Using Homoskedastic likelihood itself and using MC Dropout to capture uncertainty.
4. For point 4, do we show how the choice of kernels influences the predictions?

nipunbatra commented:
@haikookhandor please put the revised final set of images we want to use here in the issue directly. 

reg. point 4, just use 1-2 kernels. We don't need to give much details about the model and kernel anyway.
